IDO alignment

mbmaja opened the pull request:
A first draft of SOSA- IDO alignment
https://github.com/mbmaja/sdw-sosa-ssn/blob/ido-alignment/ssn/chapters/IDO-alignment.html  
see issue https://github.com/w3c/sdw-sosa-ssn/issues/383

Alignment of inverse object properties is still missing.

Alignment of the classes needs to be reviewed, alignments for sosa:ObservableProperty and sosa:ActuatableProperty probably should be removed, as these have been deprecated. Their alignment could be reflected in a different alignment of sosa:Property ( lis:Quality or lis:Disposition instead if lis:Dependant)

@mikkelbrynildsen @prylandsholm @2r5345gwa @mhodki @Caitlin2694 @aaa5nnn8

Commit:
feat: added IDO alignment draft

dr-shorthair commented:
In ./index.html in the #Alignments section add 
```
    <section id="IDO-alignment" data-include="./chapters/IDO-alignment.html"></section>
```

Then https://raw.githack.com/mbmaja/sdw-sosa-ssn/ido-alignment/ssn/index.html#IDO-alignment should show a preview

dr-shorthair commented:
@mbmaja is your institution a member of W3C? Or OGC? 

- If yes, then see https://labs.w3.org/repo-manager/pr/id/w3c/sdw-sosa-ssn/392  to clear IPR issues. 
- If no, then I think we may have to designate you an invited expert, since these are substantive additions (even if the whole Alignments chapter is strictly 'non-normative').

mbmaja commented:
@dr-shorthair My organization (Siemens) is a W3C member, however I am not a member of any W3C groups yet. 

Edit: I created a W3C account (Siemens affiliation) and joined the Spatial Data on the Web Working Group. Seems to work now.

Commit:
chore: udated index.html

dr-shorthair commented:
Preview https://raw.githack.com/mbmaja/sdw-sosa-ssn/ido-alignment/ssn/index.html#IDO-alignment highlights broken links : 

1. per #237 `sosa:ActuatableProperty` and `sosa:ObservableProperty` have been deprecated, so are no longer documented in the HTML, so the links to those are broken. The argument was that there is no intrinsic distinction between properties that can be changed and properties that can be observed, and in any control system you will observe everything you try to change. Replace with `sosa:Property`.  

But are the resulting subclass relations correct for these? Could be 
`lis:Disposition ⊑ sosa:Property` (I don't know IDO, but Disposition looks maybe more of a role than an essence)
`lis:Quality = sosa:Property` _or_ `lis:Quality ⊑ sosa:Property`

2. per #113 `sosa:Result` has also been deprecated. This is really an *any* class, and giving it a name didn't really seem to have any semantic significance. In the 2017 edition it really only had one role - to be the superclass of `sosa:Sample`. 

Personally, I feel less strongly about the deprecation of `sosa:Result`. 
If you have a real need to have a named class for all Results please make the case.

dr-shorthair commented:
We will need a citation to IDO. 
Preferably a short code from here https://www.specref.org/?q=IDO 

Go to https://github.com/tobie/specref and add it? 
I think it would be an entry in https://github.com/tobie/specref/blob/main/refs/biblio.json

Commit:
updated and completed the alignment

mbmaja commented:
@dr-shorthair I updated the alignment file and chapter. Removed the deprecated sosa classes, added missing property alignments, and updated  some of the alignments. I am not sure if I understand sosa:Property and sosa:Stimulus properly. If we can have a short call about this, it would be great.

I am also waiting for a review of my colleagues from the IDO core group.

The alignment ontology is still missing metadata. Is there any guideline about the license for alignment ontologies? Are their IRIs resolvable, if under sosa domain?

dr-shorthair commented:
`sosa:Property` is an [identifiable quality of features of interest that can be observed or acted upon](https://www.w3.org/TR/vocab-ssn-2023/#SOSAProperty). I'm pretty confident that this is the same as [lis:Quality](https://rds.posccaesar.org/ontology/lis14/rdl/Quality/). 
'feature of interest' uses the term 'feature' in the sense used in GIS. 
This commonly refers to a geographic object, though the definition is much broader and any `owl:NamedIndividual` could act as a feature of interest of an execution. 

`sosa:Stimulus` comes from https://ceur-ws.org/Vol-668/[paper12.pdf](https://ceur-ws.org/Vol-668/paper12.pdf) where it is described as "Stimuli are detectable changes in the environment, i.e., in the physical world.  They are the starting point of each measurement as they act as triggers for sensors."

As for metadata for the alignment ontology, please look at the other alignments and do your best. 
Much of this is my work and it would be really good to have another pair of eyes look over it anyway. 
Suggestions for improvement would be eagerly received. 

Finally: Please add the ref for IDO into the ssn/config.js file for now (pending your addition to specref ;-) ) so it appears in the reference list at the end of the document. 
Also add your name to the list of Contributors - in the ssn/config.js file.

Commit:
updated config.js: added IDO reference and contributor MMB

Commit:
fix: config.js

Commit:
fix: config.js

mbmaja commented:
@dr-shorthair I updated config.js but its seems to have broken the preview https://raw.githack.com/mbmaja/sdw-sosa-ssn/ido-alignment/ssn/index.html#IDO-alignment  I cannot  figure what the problem is.

dr-shorthair commented:
Please put your details in the _otherLinks/Contributors_ section (**NOT** the _editors_ section). 

And best to add ISO23726-3-DIS to SpecRef. 
The local bibliography is primarily for non-standards (e.g. papers).

dr-shorthair commented on ssn/config.js:
```diff
@@ -92,6 +92,13 @@ var respecConfig = {
       companyURL: "https://brgm.fr/",
       orcid: "0000-0001-7656-1830"
+    },
+    {
```
Please put your details in the otherLinks/Contributors section (NOT the editors section).

mbmaja replied:
done

dr-shorthair commented on ssn/config.js:
```diff
@@ -165,6 +172,12 @@ var respecConfig = {
     "Description-Logics-Home-Page": {
       href: "https://dl.kr.org/",
       title: "Description Logics Home Page"
+    },
```
ISO23726-3-DIS should be in SpecRef.
The local bibliography is primarily for non-standards (e.g. papers).

mbmaja replied:
I removed ISO23726-3-DIS from config.js and created a PR at specref

mbmaja replied:
@dr-shorthair IDO reference is available at Specref https://www.specref.org/?q=ISO23726-3-DIS

Commit:
move MMB to otherLinks/contributors

Commit:
remove IDO reference from localbib

dr-shorthair commented:
@mbmaja is this ready for review now?

dr-shorthair commented:
Need to add a short section in https://www.w3.org/TR/vocab-ssn-2023/#SSNalignmentModules

Commit:
added metadata to teh alignment ontology

mbmaja committed:
Merge pull request #1 from mbmaja/gh-pages

Rebase

Commit:
sosa-ido imoports sosa instead of ssn

mbmaja commented:
A review from a member of the IDO core group is needed, before we can remove the Draft status of this PR. I will address it next week in the weekly meeting of the IDO core group.  There are still some open questions:
-  should sosa-ido import IDO via its ontologyIRI or the current versionIRI
- should sosa:Sensor, sosa: Actuator, sosa:Sampler be subclasses of lis:PhysicalObject, considering that they can be implemented as software 
- should sosa:Observation be aligned with lis:ActivityProfile and sosa:observedProperty with lis:profileOfQuality
- should sosa:Stimulus be aligned with lis:Event, rather than lis:Activity

Commit:
added sosa-ido paragraph to Modularization.html

mbmaja commented:
@dr-shorthair 
> Need to add a short section in https://www.w3.org/TR/vocab-ssn-2023/#SSNalignmentModules

I added a section. The picture still needs to be updated (to add IDO alignment module)

Commit:
fix: section name

Commit:
Update after IDO core group review

mbmaja commented:
12th November 2025
IDO core group review was done today by @johanwk @Caitlin2694 @aaa5nnn8
The alignments are approved, the import is changed to the unversioned ontologyIRI of IDO, a note and a comment are added to the ttl file to state the version of IDO for which the alignment is made.

@dr-shorthair This PR is ready for review now - I removed the draft status. 

Note for the reviewers: in IDO, deployed/executable software is considered physical object.
Open question: how to refer to the current version of SOSA in the alignment?

dr-shorthair commented:
![SSN-alignment-modules-IDO](https://github.com/user-attachments/assets/bf57ad94-cc11-4583-a2a7-4a1ef29ff2ad)

Here's the alignments module diagram with SOSA-IDO  added.

Commit:
chore: updated image SSN-alignment-modules.svg

nicholascar commented:
There are two notes in the alignment that mention `SubClassOf` and `SubPropertyOf`, just like this, in code style. If `rdfs:subClassOf` and `rdfs:subPropertyOf` are intended, then at least the leading `S` should be made lower case and, I suggest, the `rdfs` prefix should be added.

mbmaja commented:
@nicholascar 
> There are two notes in the alignment that mention `SubClassOf` and `SubPropertyOf`, just like this, in code style. If `rdfs:subClassOf` and `rdfs:subPropertyOf` are intended, then at least the leading `S` should be made lower case and, I suggest, the `rdfs` prefix should be added.

I took over this notation from other alignment chapters. My guess is that `SubClassOf` and `SubPropertyOf` refer to Manchester OWL Syntax. I have no problem changing it to `rdfs:subClassOf` and `rdfs:subPropertyOf` , but I think it does not make sense to do it without changing the notes in the other alignment chapters in a similar way.

dr-shorthair commented:
The `Note:` about the logic notation applies to all the alignments in chapter 7, so I suggest we move it up to the top of the chapter and remove the repetition. 
 
I tend to agree that hopping out to OWL notation at his stage is a bit jarring given that we use RDF and RDFS everywhere else. But let's deal with it in one place. I'll create a new issue.

dr-shorthair commented:
@maximelefrancois86 @rob-metalinkage @oldskeptic @ldesousa could you review this so we can merge and relieve @mbmaja of her attendance from here?
